src/vs/platform/userDataProfile/electron-main/userDataProfile.ts

```diff
@@ -47,17 +47,25 @@ export class UserDataProfilesMainService extends UserDataProfilesService impleme
 		if (!this.isEnabled()) {
 			return undefined;
 		}
-		if (!args.profile) {
-			return undefined;
-		}
 		// Do not create the profile if folder/file arguments are not provided
 		if (!args._.length && !args['folder-uri'] && !args['file-uri']) {
 			return undefined;
 		}
-		if (this.profiles.some(p => p.name === args.profile)) {
-			return undefined;
+		if (args.profile) {
+			if (this.profiles.some(p => p.name === args.profile)) {
+				return undefined;
+			}
+			return this.createProfile(args.profile);
+		}
+		if (args['profile-transient']) {
+			return this.createTransientProfile()
+				.then(profile => {
+					// Set the profile name to use
+					args.profile = profile.name;
```

Comment:
Maybe we can return `Promise<NativeParsedArgs>` from `checkAndCreateProfileFromCli`?

Reply:
Will be handled in separate PR